go/inst/instance_topology.go

```diff
@@ -1675,6 +1677,33 @@ func TakeSiblings(instanceKey *InstanceKey) (instance *Instance, takenSiblings i
 	return instance, len(relocatedReplicas), err
 }
 
+// Created this function to allow a hook to be called after a successful TakeMaster event
+func TakeMasterHook(master *Instance, slave *Instance) {
+	masterInstance := master.Key
+	slaveInstance := slave.Key
+	env := goos.Environ()
+	env = append(env, fmt.Sprintf("ORC_SUCCESSOR_HOST=%s", masterInstance))
+	env = append(env, fmt.Sprintf("ORC_FAILED_HOST=%s", slaveInstance))
+	successor := fmt.Sprintf("%s", masterInstance)
+	oldmaster := fmt.Sprintf("%s", slaveInstance)
+	if config.Config.PostTakeMasterProcesses != nil {
```

Comment:
I'm wondering why `PostTakeMasterProcesses` is needed. If you don't want to have `TakeMaster` hooks, just make `PostTakeMasterProcesses` empty, right?

Reply:
Thanks Shlomi for reviewing all of this. I'll remove the PostTakeMasterProcessesEnabled from the config and instance_topology.go (I think that's what you meant), that way, as you have said, if no hook is present then it'll be empty and not called.

Reply:
I've pushed the changes, please let me know if this is ok. Thanks for all the help and input 😄  👍